### DISCUSSION TITLE
"The operation has timed out" when connection to IVoiceChannel

### DISCUSSION THREAD

**Discussion by KuzmichMichael** [discord-net/Discord.Net#2641] [Q&A] [answered]
Hello everyone.

I recently worked on a music bot for discord and it worked well for about a month. But about a week ago he stopped connecting to voice channels with a timeout. I tried connecting to different channels on different servers, but it didn't help.

Here is my code
```
        [Discord.Commands.RequireContext(ContextType.Guild, ErrorMessage = _err)]
        [SlashCommand("join", "Join voice channel")]
        public async Task JoinAsync()
        {
            await AudioServices.JoinAsync((Context.User as IGuildUser)?.VoiceChannel);
        }
```

```
        public async Task JoinAsync(IVoiceChannel voiceChannel)
        {
            await voiceChannel.ConnectAsync();
        }
```

![error](https://user-images.githubusercontent.com/81437501/228226806-691c9ff7-c12d-47f4-8905-e2fda736bc4a.png)

Has anyone encountered this problem or knows its solution?

**Comment by Misha-133** [comment, marked as answer]
Upgrade D.Net to 3.10

**Reply by KuzmichMichael** [reply]
It helped. Thank you very much. For some reason I thought that I already had version 3.10

**Reply by rillyboss** [reply]
Im still seeing this issue with 3.10. Are there any configuration items I may be missing? The bot joins voice and then disconnects shortly after.
